crates/shim/src/mount.rs

```diff
@@ -562,8 +562,8 @@ pub fn mount_rootfs(
         None
     };
 
-    unshare(CloneFlags::CLONE_FS).unwrap();
     if let Some(workdir) = chdir {
+        unshare(CloneFlags::CLONE_FS).unwrap();
```

Comment:
Would you please add UT for that? Thanks

Reply:
Yes, I will do it later.

Reply:
Done.